things/model/src/main/java/org/eclipse/ditto/things/model/signals/events/ThingMigrated.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.things.model.signals.events;
+
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.base.model.entity.metadata.Metadata;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.FieldType;
+import org.eclipse.ditto.base.model.json.JsonParsableEvent;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.base.model.signals.UnsupportedSchemaVersionException;
+import org.eclipse.ditto.base.model.signals.commands.Command;
+import org.eclipse.ditto.base.model.signals.events.EventJsonDeserializer;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.json.JsonValue;
+import org.eclipse.ditto.things.model.Thing;
+import org.eclipse.ditto.things.model.ThingsModelFactory;
+
+/**
+ * This event is emitted after a {@link org.eclipse.ditto.things.model.Thing} was successfully migrated.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+@JsonParsableEvent(name = ThingMigrated.NAME, typePrefix = ThingEvent.TYPE_PREFIX)
+public final class ThingMigrated extends AbstractThingEvent<ThingMigrated> implements ThingModifiedEvent<ThingMigrated> {
+
+    /**
+     * Name of the "Thing Migrated" event.
+     */
+    public static final String NAME = "thingMigrated";
+
+    /**
+     * Type of this event.
+     */
+    public static final String TYPE = TYPE_PREFIX + NAME;
+
+    private final Thing thing;
+
+    private ThingMigrated(final Thing thing,
+                          final long revision,
+                          @Nullable final Instant timestamp,
+                          final DittoHeaders dittoHeaders,
+                          @Nullable final Metadata metadata) {
+        super(TYPE, thing.getEntityId().orElseThrow(() -> new NullPointerException("Thing has no ID!")), revision, timestamp, dittoHeaders, metadata);
```

Comment:
I would not throw `NullPointerException` - but rather `IllegalArgumentException`.
NPE is something really technical.